test/Main.purs

```diff
@@ -110,11 +110,17 @@ main = runTest do
         , "+3.50"
         ]
 
+      allParseAs (Expression (scalar 0.2))
+        [ "0.2"
+        , "  0.2  "
+        , "+0.2 "
+        , ".2"
+        ]
```

Comment:
Before I take a closer look at the implementation: can we think edge cases that we need to test? For now, we have just one test for the new syntax (`".2"`). Do we also want to test things like `"+.2"`, `"-.2"`, `".02"` and add a few things that we *don't* want to be parsed as a number (e.g. `". 2"`)?